packages/angular/build/src/tools/esbuild/application-code-bundle.ts

```diff
@@ -570,6 +571,12 @@ function getEsBuildCommonOptions(options: NormalizedApplicationBuildOptions): Bu
     'es2015',
     'es2020',
   ];
+  // JIT requires additional metadata not included in the pre-linked code
+  if (!jit) {
```

Comment:
The linker does support preserving JIT code if I remember correctly. Should the APF code run with that mode?

Reply:
JIT is currently setup to link at runtime with the CLI so i would avoid the extra code for the common AOT case.